data-prepper-plugins/s3-source/src/test/java/org/opensearch/dataprepper/plugins/source/S3SelectResponseHandlerTest.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.source;
+
+import org.junit.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.core.async.SdkPublisher;
+import software.amazon.awssdk.services.s3.model.Progress;
+import software.amazon.awssdk.services.s3.model.RecordsEvent;
+import software.amazon.awssdk.services.s3.model.SelectObjectContentEventStream;
+import software.amazon.awssdk.services.s3.model.SelectObjectContentResponse;
+import software.amazon.awssdk.services.s3.model.Stats;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.SequenceInputStream;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.CoreMatchers.sameInstance;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.mock;
+
+public class S3SelectResponseHandlerTest {
+    @Test
+    public void testS3SelectResponseHandlerWithMockedResponse() throws IOException {
+        S3SelectResponseHandler responseHandler = new S3SelectResponseHandler();
+        final SdkPublisher publisher = mock(SdkPublisher.class);
+        responseHandler.onEventStream(publisher);
+        responseHandler.responseReceived(mock(SelectObjectContentResponse.class));
+        responseHandler.complete();
+        final List<SelectObjectContentEventStream> receivedEvents =
+                responseHandler.getS3SelectContentEvents();
+        assertNotNull(receivedEvents);
+    }
+
+    @Test
+    public void testS3SelectResponseHandler() throws IOException {
+        S3SelectResponseHandler responseHandler = new S3SelectResponseHandler();
+        final RuntimeException exception = mock(RuntimeException.class);
+        responseHandler.exceptionOccurred(exception);
+        responseHandler.complete();
+        final List<SelectObjectContentEventStream> receivedEvents =
+                responseHandler.getS3SelectContentEvents();
+        assertNotNull(receivedEvents);
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings= {"{\"name\":\"Test\"}",
+            "{\"S.No\":\"1\",\"name\":\"data-prep\",\"country\":\"USA\"}",
+            "{\"S.No\":\"6\",\"name\":\"data-prep-test\",\"type\":\"json\"}",
+            "{\"S.No\":\"5\",\"log\":\"data-prep-log\",\"documentType\":\"test doc\"}",
+            "{\"S.No\":\"4\",\"name\":\"data-prep\",\"empId\",\"123456\"}"})
+    public void testS3SelectResponseHandlerWithMultipleEvents(final String response) throws IOException {
```

Comment:
This isn't exactly what I meant by handling multiple events.

The S3 SDK may call the `responseReceived` method multiple times. What I'd like to see is a test which verifies that multiple calls to `responseReceived` work as expected.

Here is what it would look like. I didn't mock the objects here. But, this should convey the idea.

```
responseHandler.responseReceived(object1);
responseHandler.responseReceived(object2);
responseHandler.responseReceived(object3);

List<SelectObjectContentEventStream> receivedEvents = responseHandler.getS3SelectContentEvents();
assertThat(receivedEvents, notNullValue());
assertThat(receivedEvents.size(), equalTo(3));
assertThat(receivedEvents.get(0), equalTo(object1));
assertThat(receivedEvents.get(1), equalTo(object2));
assertThat(receivedEvents.get(2), equalTo(object3));
```

Reply:
Thanks David for your review, Events are not receiving from responseReceived method, Actually receiving event records from **onEventStream(...)**. So we have added test case accordingly. Please verify and let us know your feedback.